### DISCUSSION TITLE
Recheck on save

### DISCUSSION THREAD

**Discussion by andrewbaxter** [microsoft/vscode-mypy#270] [Q&A]
The readme says

> Mypy will be automatically executed when you open a Python file

I can confirm that upon opening lint errors are displayed.  When I save the file they all disappear and never come back until I re-open the file or restart mypy. That's not what I expected, but it doesn't contradict the stated behavior.

Is there a way to get it to re-check without restarting or re-opening the file? Or more specifically, is there a way to get it to recheck every time I save (or even while I'm typing, with a delay maybe)?

Or is there some issue with my installation?

**Comment by karthiknadig** [comment]
@andrewbaxter If you are running `dmypy` then it might be issue with `dmypy` caching (this is a reported problem on `dmypy` see https://github.com/python/mypy/issues/15677). So, try changing that to `mypy` using `mypy-type-checker.preferDaemon` to `false`.

You can also look at Output > Mypy Type Checker, to see if it triggers `mypy` on save.
